Failed to Initialize Gradle Language Server

Extension Name: vscode-gradle
Extension Version: v3.8.0
OS Version: Windows 10.0.19043 Build 19043
VSCode version: 1.60.2

**Describe the bug**
Upon opening a Gradle project, the Gradle Language Server throws errors and fails to start properly. The status bar indicates it's stuck in "Initializing Gradle Language Server". Upon opening a *.java file or the Tasks panel, the Language Server tries again to start up and succeeds.

**To Reproduce**
Open any folder with a build.gradle at the root (including this repository itself).

**Expected behavior**
No error should be thrown. Status bar should not be stuck with "Initializing Gradle Language Server" spinner.

**Output from "Gradle Tasks"**
```
[debug] [gradle-server] Starting server
[debug] [gradle-server] Gradle Server cmd: c:\Users\shaun_000\.vscode\extensions\vscjava.vscode-gradle-3.8.0\lib\gradle-server.bat 49258
[info] [gradle-server] Server started, listening on 49258
[info] Gradle client connected to server
```

**Output from "Gradle Language Server"**
```
Oct 05, 2021 9:53:28 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: java.lang.reflect.InvocationTargetException
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: java.lang.NullPointerException: Cannot read the array length because "<local3>" is null
	at com.microsoft.gradle.resolver.GradleLibraryResolver.searchInFolder(GradleLibraryResolver.java:161)
	at com.microsoft.gradle.resolver.GradleLibraryResolver.findCoreAPIWithDist(GradleLibraryResolver.java:152)
	at com.microsoft.gradle.resolver.GradleLibraryResolver.findCoreAPIWithWrapper(GradleLibraryResolver.java:140)
	at com.microsoft.gradle.resolver.GradleLibraryResolver.resolveGradleAPI(GradleLibraryResolver.java:89)
	at com.microsoft.gradle.GradleServices.applySetting(GradleServices.java:160)
	at com.microsoft.gradle.GradleLanguageServer.initialize(GradleLanguageServer.java:90)
	... 16 more

[Error - 9:53:29 PM] Server initialization failed.
  Message: Internal error.
  Code: -32603 
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: java.lang.NullPointerException: Cannot read the array length because "<local3>" is null
	at com.microsoft.gradle.resolver.GradleLibraryResolver.searchInFolder(GradleLibraryResolver.java:161)
	at com.microsoft.gradle.resolver.GradleLibraryResolver.findCoreAPIWithDist(GradleLibraryResolver.java:152)
	at com.microsoft.gradle.resolver.GradleLibraryResolver.findCoreAPIWithWrapper(GradleLibraryResolver.java:140)
	at com.microsoft.gradle.resolver.GradleLibraryResolver.resolveGradleAPI(GradleLibraryResolver.java:89)
	at com.microsoft.gradle.GradleServices.applySetting(GradleServices.java:160)
	at com.microsoft.gradle.GradleLanguageServer.initialize(GradleLanguageServer.java:90)
	... 16 more

[Error - 9:53:29 PM] Starting client failed
  Message: Internal error.
  Code: -32603 
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: java.lang.NullPointerException: Cannot read the array length because "<local3>" is null
	at com.microsoft.gradle.resolver.GradleLibraryResolver.searchInFolder(GradleLibraryResolver.java:161)
	at com.microsoft.gradle.resolver.GradleLibraryResolver.findCoreAPIWithDist(GradleLibraryResolver.java:152)
	at com.microsoft.gradle.resolver.GradleLibraryResolver.findCoreAPIWithWrapper(GradleLibraryResolver.java:140)
	at com.microsoft.gradle.resolver.GradleLibraryResolver.resolveGradleAPI(GradleLibraryResolver.java:89)
	at com.microsoft.gradle.GradleServices.applySetting(GradleServices.java:160)
	at com.microsoft.gradle.GradleLanguageServer.initialize(GradleLanguageServer.java:90)
	... 16 more
```

**Does the bug still exist if you disable all other extensions?**
Yes

**Additional context**
After waiting for a couple minutes after opening the `vscode-gradle` repository folder, the following errors popped up. I haven't seen these before so I suspect they may be related to the vscode-gradle repository itself rather than this issue (which applies to all gradle projects) but including them just in case:
![image](https://user-images.githubusercontent.com/7451023/136144151-8c1112c5-e64c-4f35-9151-c08e52633a05.png)

--- a/gradle-language-server/src/main/java/com/microsoft/gradle/resolver/GradleLibraryResolver.java
+++ b/gradle-language-server/src/main/java/com/microsoft/gradle/resolver/GradleLibraryResolver.java
@@ -149,12 +149,20 @@ public class GradleLibraryResolver {
 
   private File findCoreAPIWithDist(Path gradleUserHomePath, String gradleDist) {
     Path distPath = gradleUserHomePath.resolve(Path.of("wrapper", "dists"));
-    File distFolder = searchInFolder(gradleDist, distPath.toFile());
-    if (distFolder != null && distFolder.exists()) {
-      Path libPath = distFolder.toPath().resolve("lib");
-      return findCoreAPI(libPath.toFile());
+    File distFolder = distPath.toFile();
+    if (!isValidFolder(distFolder)) {
+      return null;
     }
-    return null;
+    File targetFolder = searchInFolder(gradleDist, distFolder);
+    if (!isValidFolder(targetFolder)) {
+      return null;
+    }
+    Path libPath = targetFolder.toPath().resolve("lib");
+    File libFolder = libPath.toFile();
+    if (!isValidFolder(libFolder)) {
+      return null;
+    }
+    return findCoreAPI(libFolder);
   }
 
   private File searchInFolder(String gradleDist, File folder) {
@@ -251,5 +259,9 @@ public class GradleLibraryResolver {
 
   private static boolean isValidFile(File file) {
     return file != null && file.exists();
+  }
+
+  private static boolean isValidFolder(File folder) {
+    return folder != null && folder.exists() && folder.isDirectory();
   }
 }

